### PR TITLE
Include original task ID in checkpoint info response

### PR DESCRIPTION
**Pull request by pawalt**
## Describe your changes

This commit adds the task ID which creategd the checkpoint to CheckpointInfo. This allows us to restore a container with the exact same arguments which were used to create it.

<details> <summary>Backward/forward compatibility checks</summary>

---

Check these boxes or delete any item (or this section) if not relevant for this PR.

- [x] Client+Server: this change is compatible with old servers
- [x] Client forward compatibility: this change ensures client can accept data intended for later versions of itself

Note on protobuf: protobuf message changes in one place may have impact to
multiple entities (client, server, worker, database). See points above.

---

</details>

### CODE REVIEW

File: modal_proto/api.proto [hunk 1]
```diff
@@ -588,7 +588,8 @@ message CheckpointInfo {
   string checkpoint_id = 3;
   string runtime_fingerprint = 4;
   int64 size = 5;
-  bool checksum_is_file_index = 6;
+  bool checksum_is_file_index = 6;
```
Comment (pawalt) [inline modal_proto/api.proto:591]: I'm not sure why this is a diff

File: modal_proto/api.proto [hunk 2]
```diff
@@ -589,6 +589,7 @@ message CheckpointInfo {
   string runtime_fingerprint = 4;
   int64 size = 5;
   bool checksum_is_file_index = 6;
+  string task_id = 7;
```
Comment (cweld510) [inline modal_proto/api.proto:592]: Can we call this `original_task_id` or something else that makes it clear which task it refers to?

**Review by cweld510 — APPROVED** [review]

File: modal_proto/api.proto (hunk 2 above)
Comment (pawalt) [reply modal_proto/api.proto]: ya

**Comment by pawalt** [comment]
@prbot automerge

**Pull request merged**